components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/jwt/JWKSBasedJWTValidator.java

```diff
@@ -219,4 +236,32 @@ private void setJWKeySelector(String jwksUri, String algorithm) throws Malformed
         JWSKeySelector<SecurityContext> keySelector = new JWSVerificationKeySelector<>(expectedJWSAlg, keySource);
         jwtProcessor.setJWSKeySelector(keySelector);
     }
+
+    /**
+     * Validate the certificate based on the signature algorithm.
+     *
+     * @param x509Certificate X509Certificate
+     * @param alg             Signature Algorithm
+     * @return true if the certificate is valid.
+     */
+    private boolean isValidCertificate(X509Certificate x509Certificate, String alg) {
+
+        if (alg.startsWith(RS) || alg.startsWith(PS)) {
+            // At this point 'x509Certificate' will never be null.
+            PublicKey publicKey = x509Certificate.getPublicKey();
+            if (publicKey instanceof RSAPublicKey) {
+                return true;
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Public key is not an RSA public key.");
+                }
+                return false;
+            }
```

Comment:
lets keep this content in isValidCertificate method. and move the rest to a validate signingAlgorithm method.

Reply:
Fixed with: https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2701/commits/1491ff1df58cbcb49d62356f8237e1da6f59afe6